kotlinx-coroutines-core/common/src/JobSupport.kt

```diff
@@ -294,11 +295,10 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
             state.list?.notifyCompletion(cause)
         }
         /*
-         * 4) Invoke onCompletionInternal: onNext(), timeout de-registration etc.
-         *    It should be last so all callbacks observe consistent state
-         *    of the job which doesn't depend on callback scheduling.
+         * 3) Resumes the rest of the code in scoped coroutines
+         *    (runBlocking, coroutineScope, withContext, withTimeout, etc)
          */
-        onCompletionInternal(update, mode)
+        afterCompletionInternal(update, mode)
```

Comment:
Why some functions have "internal" suffix, while others haven't?

Reply:
This is a legacy distinction from the times we had "internal" `JobSupport` and "public" `AbstractCoroutine`. Functions with `Internal` suffix were never supposed to become part of public API. They are either taking `mode: Int` or operating on `state: Any` (or both).